packages/core/src/config/load.ts

```diff
@@ -120,25 +135,106 @@ export function getLintRawConfigWithMergedContentByPriority(
+): Promise<Record<string, Api>> {
+  const apiKeys = Object.keys(apis);
+  if (isEmptyArray(apiKeys)) {
+    return apis;
+  }
+
+  const resolvedApis: Record<string, Api> = {};
+  for (const apiKey of apiKeys) {
+    if (!apis[apiKey]?.lint?.extends) {
+      resolvedApis[apiKey] = apis[apiKey];
+    }
+    const lint = await resolveExtends(apis[apiKey].lint as LintRawConfig, configPath, resolve);
+    resolvedApis[apiKey] = {...apis[apiKey], lint};
+  }
+
+  return resolvedApis;
+}
+
+async function resolveExtends(
+  lintConfig: LintRawConfig | Omit<LintRawConfig, "plugins">,
+  configPath: string = '',
+  resolve?: RawResolveConfig,
+): Promise<LintRawConfig> {
+  if (!lintConfig.extends || isEmptyArray(lintConfig.extends)) {
+    return lintConfig;
+  }
+
   if (lintConfig.extends.some(isNotString)) {
-    throw Error(`Error configuration format not detected in lint.extends`); // TODO: show correct errors
+    throw new Error(`Error configuration format not detected in extends value must contain strings`);
   }
 
-  const lintExtends = await Promise.all(
-    lintConfig.extends.map(async (item) =>
-      isAbsoluteUrl(item) || fs.existsSync(item)
-        ? loadExtendLintConfig(item).then(resolveExtends)
-        : item,
-    ),
-  );
-  // TODO: check perf. - if lintExtends contains only strings, we can simply return lintConfig
-  return getLintRawConfigWithMergedContentByPriority({ ...lintConfig, extends: lintExtends });
+  const lintExtends: (string | LintRawConfig)[] = [];
+  const pluginExtends: (string | Plugin)[] = [];
+  for (const item of lintConfig.extends) {
+    if (!isAbsoluteUrl(item) && !fs.existsSync(item)) {
+      lintExtends.push(item);
+      continue;
+    }
+
+    const extendedLintConfig = await loadExtendLintConfig(item, resolve);
+
+    if (extendedLintConfig.plugins && !isEmptyArray(extendedLintConfig.plugins)) {
+      const plugins = extendedLintConfig.plugins
+        .map((plugin) => resolveNestedPlugins(configPath, item, plugin))
+        .filter(Boolean) as (string | Plugin)[];
+      pluginExtends.push(...plugins);
+    }
+
+    if (extendedLintConfig.extends) {
+      lintExtends.push(await resolveExtends(extendedLintConfig, configPath, resolve));
+      continue;
+    }
+
+    lintExtends.push(extendedLintConfig);
+  }
+
+  return getLintRawConfigWithMergedContentByPriority({ ...lintConfig, extends: lintExtends, plugins: pluginExtends });
 }
 
-async function loadExtendLintConfig(filePath: string): Promise<LintRawConfig> {
-  // TODO: should test urls and handle errors
-  const fileSource = await new BaseResolver().loadExternalRef(filePath);
-  return (parseYaml(fileSource.body) as RawConfig).lint || {};
+async function loadExtendLintConfig(
+  filePath: string,
+  resolve?: RawResolveConfig,
+): Promise<LintRawConfig> {
+  try {
+    const fileSource = await new BaseResolver(getResolveConfig(resolve)).loadExternalRef(filePath);
+    const rawConfig = parseYaml(fileSource.body) as RawConfig;
```

Comment:
We probably need to use `transformConfig` adapter somewhere here to be sure we're working with the new `RawConfig` and not `DeprecatedRawConfig`.